### DISCUSSION TITLE
replace php constant index to named index

### DISCUSSION THREAD

**Discussion by rakucmr** [rectorphp/rector#8279] [Q&A] [answered]
In php 4, and php 5 a constant if not found is automatically converted to string so a code like this is ok:

`$array[constant_index]`

is equivalent to

`$array['constant_index']`

How to use rector to convert these types of indexes to work correctly in php 7 / 8?

Thanks!

**Comment by samsonasik** [comment, marked as answer]
I think you can create custom rule for that, as creating rule for core is too risky for core:

- You need to ensure all constants loaded via `$rectorConfig->bootstrapFiles()`, ensure that it actually not a constant
- then, change the ArrayDimFetch, if its dim is a ConstFetch that not exists via :

```php
if ($this->reflectionProvider->hasConstant(new Name($constantName), $scope)) {
```

If not, change to string, you can check rector's `SensitiveConstantNameRector` which change to case sensitive constant name instead:

https://github.com/rectorphp/rector/blob/ae4c0b72cba90c260fccbc85004e5ceb2bb47dfe/rules/Php73/Rector/ConstFetch/SensitiveConstantNameRector.php#L78-L80

and create custom rule based on your needs.
